kornia/losses/lovasz_hinge.py

```diff
@@ -0,0 +1,149 @@
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+
+# based on:
+# https://github.com/bermanmaxim/LovaszSoftmax
+
+
+def lovasz_hinge_loss(input: torch.Tensor, target: torch.Tensor) -> torch.Tensor:
+    r"""Criterion that computes a surrogate binary intersection-over-union (IoU) loss.
+
+    According to [2], we compute the IoU as follows:
+
+    .. math::
+
+        \text{IoU}(x, class) = \frac{|X \cap Y|}{|X \cup Y|}
+
+    [1] approximates this fomular with a surrogate, which is fully differentable.
+
+    Where:
+       - :math:`X` expects to be the scores of each class.
+       - :math:`Y` expects to be the binary tensor with the class labels.
+
+    the loss, is finally computed as:
+
+    .. math::
+
+        \text{loss}(x, class) = 1 - \text{IoU}(x, class)
+
+    Reference:
+        [1] http://proceedings.mlr.press/v37/yub15.pdf
+        [2] https://arxiv.org/pdf/1705.08790.pdf
+
+    . note::
+        This loss function only supports binary labels. For multi-class labels please
+        use the Lovasz-Softmax loss.
+
+    Args:
+        input: logits tensor with shape :math:`(N, 1, H, W)`.
+        labels: labels tensor with shape :math:`(N, H, W)` with binary values.
+
+    Return:
+        a scalar with the computed loss.
+
+    Example:
+        >>> N = 1  # num_classes
+        >>> input = torch.randn(1, N, 3, 5, requires_grad=True)
+        >>> target = torch.empty(1, 3, 5, dtype=torch.long).random_(N)
+        >>> output = lovasz_hinge_loss(input, target)
+        >>> output.backward()
+    """
+    if not isinstance(input, torch.Tensor):
+        raise TypeError(f"Input type is not a torch.Tensor. Got {type(input)}")
+
+    if not isinstance(target, torch.Tensor):
+        raise TypeError(f"Target type is not a torch.Tensor. Got {type(target)}")
+
+    if not len(input.shape) == 4:
+        raise ValueError(f"Invalid input shape, we expect Bx1xHxW. Got: {input.shape}")
+
+    if not len(target.shape) == 3:
+        raise ValueError(f"Invalid target shape, we expect BxHxW. Got: {target.shape}")
+
+    if not input.shape[1] == 1:
+        raise ValueError(f"Invalid input shape, we expect Bx1xHxW. Got: {input.shape}")
+
+    if not input.shape[-2:] == target.shape[-2:]:
+        raise ValueError(f"input and target shapes must be the same. Got: {input.shape} and {target.shape}")
+
+    if not input.device == target.device:
+        raise ValueError(f"input and target must be in the same device. Got: {input.device} and {target.device}")
+
+    # flatten input and target [B, -1] and to float
+    input_flatten: torch.Tensor = input.flatten(start_dim=1)
```

Comment:
```suggestion
    input_flatten: Tensor = input.reshape(B, -1)
```
1. We are trying to isolate torch imports to make the api more pythonic and generic potentially to other frameworks. I'd recommend to `from torch import Tensor` and possibly reduce the Tensor typings to make more readable the code. Specially for losses which are mostly tensors and includes mathematical formulation.
2. Use `.reshape()` since it's used across different frameworks, like numpy, jax, etc

Reply:
What's about the .view() call, should I go also for .reshape() there? If not done!